clevertap-core/src/main/java/com/clevertap/android/sdk/inapp/InAppController.java

```diff
@@ -851,7 +857,23 @@ private static void showInApp(Context context, final CTInAppNotification inAppNo
                 break;
             case CTInAppTypeCustomCodeTemplate:
                 //TODO CustomTemplates download all file arguments before presenting
-                inAppController.presentTemplate(inAppNotification);
+                final List<String> fileUrls = inAppNotification.getCustomTemplateData()
+                        .getFileArgsUrls(inAppController.getTemplatesManager());
+                FileResourcesRepoImpl assetRepo = FileResourcesRepoFactory.createFileResourcesRepo(context,
+                        config.getLogger(),
+                        inAppController.getStoreRegistry());
+                if (assetRepo != null && !fileUrls.isEmpty()) {
+                    assetRepo.fetchAllFiles(fileUrls, (aBoolean, stringBooleanMap) -> {
+                        config.getLogger().verbose(config.getAccountId(),
+                                "file download status from showInApp() = " + aBoolean + " and url status map = "
+                                        + stringBooleanMap);
+                        if (aBoolean)
+                        {
+                            inAppController.presentTemplate(inAppNotification);
+                        }
+                        return null;
+                    });
+                }
```

Comment:
we should check if file required to present this template has finished download. 

Will this block wait for all resources to finish downloading?

Reply:
resolved